pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -197,7 +209,20 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 				reqLogger.Error(err, "could not create a new Secret")
 				return reconcile.Result{}, err
 			}
+			// Set the endpoint secret name and update .Spec.Security
 			infinispan.Spec.Security.EndpointSecretName = secret.GetName()
+			err = r.client.Update(context.TODO(), infinispan)
```

Comment:
@rigazilla what do you think about to move this update whole infinispan and Security into the separate function as looks like this is common code for the update?

Reply:
That could be fine, but first I would like to find a solution to this kind of errors:
https://travis-ci.org/infinispan/infinispan-operator/jobs/644097070#L925
we have a lot of them. I though it was enough to reload the object before apply the update, but it doesn't seem the definitive solution.
The error above comes from here:
https://github.com/rigazilla/infinispan-operator/blob/71cefaba4a2dff40a050045005e482352bd532a6/pkg/controller/infinispan/infinispan_controller.go#L325
and in this case the infinispan object is reloaded just two lines above.

Reply:
Uh no, in this case the actual error was here:
https://github.com/rigazilla/infinispan-operator/blob/71cefaba4a2dff40a050045005e482352bd532a6/test/e2e/util/test_kubernetes.go#L179